### PR TITLE
Protocol V1 E2E test setup

### PR DESCRIPTION
**Pull request by edgao**
This branch is a minimal change on top of https://github.com/airbytehq/airbyte/pull/20036 to support e2e testing. It just sets the normalization image tag to `protocolv1` for all destinations (this is equivalent to what we'll eventually do, which is publish a new version `0.3.0` and update all connectors to use 0.3.0).

Test setup process:
```shell
git checkout etsybaev/17647-gugquery-to-handle-new-datatypes
git pull
./gradlew clean :airbyte-integrations:connectors:destination-bigquery:airbyteDocker --rerun-tasks
docker tag airbyte/destination-bigquery:dev airbyte/destination-bigquery:protocolv1

git checkout edgao/protocolv1/destination-s3
git pull
./gradlew clean :airbyte-integrations:connectors:destination-s3:airbyteDocker --rerun-tasks
docker tag airbyte/destination-s3:dev airbyte/destination-s3:protocolv1

git checkout postgres_data_type_upgrade
git pull
./gradlew clean :airbyte-integrations:connectors:source-postgres:airbyteDocker --rerun-tasks
docker tag airbyte/source-postgres:dev airbyte/source-postgres:protocolv1

git checkout edgao/protocolv1/e2e_testing
git pull
# Build normalization images
./gradlew clean :airbyte-integrations:bases:base-normalization:airbyteDocker --rerun-tasks
docker tag airbyte/normalization:dev airbyte/normalization:protocolv1
docker tag airbyte/normalization-clickhouse:dev airbyte/normalization-clickhouse:protocolv1
docker tag airbyte/normalization-mssql:dev airbyte/normalization-mssql:protocolv1
docker tag airbyte/normalization-mysql:dev airbyte/normalization-mysql:protocolv1
docker tag airbyte/normalization-oracle:dev airbyte/normalization-oracle:protocolv1
docker tag airbyte/normalization-redshift:dev airbyte/normalization-redshift:protocolv1
docker tag airbyte/normalization-snowflake:dev airbyte/normalization-snowflake:protocolv1
docker tag airbyte/normalization-tidb:dev airbyte/normalization-tidb:protocolv1
# Build platform images
VERSION=dev_protocolv1 SUB_BUILD=PLATFORM ./gradlew build -x test

# Start platform
VERSION=dev_protocolv1 docker-compose up
```

This will get you an V1-enabled OSS instance, still using v0 connectors. destination-bigquery, destination-s3, and source-postgres will have a `protocolv1` tag available.

Related PRs:
* https://github.com/airbytehq/airbyte/pull/20036 - the base protocolv1 branch. All protocolv1 PRs are based on this branch.
* https://github.com/airbytehq/airbyte/pull/19403 - source-postgres and related changes
* https://github.com/airbytehq/airbyte/pull/20898 - destination-bigquery
* https://github.com/airbytehq/airbyte/pull/21358 - destination-s3 (just updates the spec; actual code changes are already available on master)

### CODE REVIEW

**Comment by edgao** [comment]
Connector versions under test:
* v0 connectors:
    * `source-postgres:1.0.36`
    * `destination-bigquery:1.2.9`
    * `destination-snowflake:0.4.42`
    * `destination-s3:0.3.18`
* v1 connectors:
    * `source-postgres:protocolv1`
    * `destination-bigquery:procotolv1`
    * `destination-s3:protocolv1`

# fresh connections; platform + normalization on protocol v1; connectors on v0

In this configuration, it doesn't matter which source/destination we use - they're all going through the v0 -> v1 -> v0 migration path. Expected behaviors:
* `NaN` shows up as `null` (because protocolv0 doesn't support nan/infinity)
* binary data is discovered as a string (source-postgres:1.0.36 doesn't have support for binary data)
* connection and actor_catalog contain v1 catalog blobs (because platform only works with v1 objects)

tl;dr these all happened correctly

**Comment by edgao** [comment]
## Upgrade destinations to v1 + sync

✅  This should work exactly the same

# v0 platform + v0 normalization + v1 catalog

this represents if a user upgrades platform, runs a sync, and then downgrades platform _without_ also downgrading their catalog. I.e. this shouldn't happen, but we should do something non-ridiculous.

## With v0 connectors
❌  Sync is successful
> ✅ snowflake and s3 both work correctly
> ❌ bigquery crashes in normalization due to [this](https://github.com/airbytehq/airbyte/blob/1b52c30ea58d979eb120c3f50ce6d437044b0f5c/airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py#L802) line

❔ Data gets normalized in a reasonable format (probably `jsonb` or equivalent)

## With v1 connectors
❔ Sync is successful
❔ Data gets normalized in a reasonable format (probably `jsonb` or equivalent), except in destination-s3 (which probably still handles everything correctly?)
